ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/LgrDB.hs

```diff
@@ -460,10 +462,17 @@ streamAPI ::
      forall m blk.
      (IOLike m, HasHeader blk)
   => ImmutableDB m blk -> StreamAPI m blk
-streamAPI immutableDB = StreamAPI streamAfter
+streamAPI = streamAPI' (const $ return False)
+
+streamAPI' ::
+     forall m blk b.
+     (IOLike m, HasHeader blk)
+  => (blk -> m Bool) -- ^ Stop condition
```

Comment:
I wonder if we want to use a custom data type for the stopping condition, eg `data ShouldStop = Stop | Proceed`.